### DISCUSSION TITLE
How to host Streamsync application on Windows 10

### DISCUSSION THREAD

**Discussion by HamsterSqueak** [writer/writer-framework#181] [General]
I am investigating using Streamsync to provide a UI for an application over a Local Area Network. I have attempted to follow the directions given for using uvicorn. I am using Visual Studio. When I run the application, I receive the following dump:

ERROR: Traceback (most recent call last): File "E:\4users\Hamilton\dev\Python\GUI\Streamsync\ssServer\ssServer\ssServerenv\Lib\site-packages\starlette\routing.py", line 677, in lifespan async with self.lifespan_context(app) as maybe_state: File "C:\Python311\Lib\contextlib.py", line 204, in aenter return await anext(self.gen) ^^^^^^^^^^^^^^^^^^^^^ File "E:\4users\Hamilton\dev\Python\GUI\Streamsync\ssServer\ssServer\ssServerenv\Lib\site-packages\streamsync\serve.py", line 42, in lifespan app_runner.load() File "E:\4users\Hamilton\dev\Python\GUI\Streamsync\ssServer\ssServer\ssServerenv\Lib\site-packages\streamsync\app_runner.py", line 595, in load self._start_app_process() File "E:\4users\Hamilton\dev\Python\GUI\Streamsync\ssServer\ssServer\ssServerenv\Lib\site-packages\streamsync\app_runner.py", line 759, in _start_app_process self.app_process.start() File "C:\Python311\Lib\multiprocessing\process.py", line 121, in start self._popen = self._Popen(self) ^^^^^^^^^^^^^^^^^ File "C:\Python311\Lib\multiprocessing\context.py", line 224, in _Popen return _default_context.get_context().Process._Popen(process_obj) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "C:\Python311\Lib\multiprocessing\context.py", line 336, in _Popen return Popen(process_obj) ^^^^^^^^^^^^^^^^^^ File "C:\Python311\Lib\multiprocessing\popen_spawn_win32.py", line 45, in init prep_data = spawn.get_preparation_data(process_obj._name) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "C:\Python311\Lib\multiprocessing\spawn.py", line 158, in get_preparation_data _check_not_importing_main() File "C:\Python311\Lib\multiprocessing\spawn.py", line 138, in _check_not_importing_main raise RuntimeError(''' RuntimeError: An attempt has been made to start a new process before the current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.

    ERROR: Application startup failed. Exiting.

Can anyone suggest how to host a Streamsync application on a Local Area Network?

**Comment by FabienArcellier** [comment]
How do you run the application on vscode to get this error message ? Could you share your configuration ([the yaml file](https://github.com/FabienArcellier/blueprint-streamsync/blob/master/.vscode/launch.json))

---
Did you try to host the streamsync application through docker ? https://www.streamsync.cloud/deploy-with-docker.html

You can then run the application in docker container and expose it on your local network.

**Reply by HamsterSqueak** [reply]
I did not try docker. I saw some instructions for using docker. I have never used docker, so I was looking for another solution. I can certainly use docker, if that solution works. I did read somewhere very recently that Streamsync requires an internet connection. If that is the case, I'm not sure that will work for the projects that I have in mind. I used the locally hosted "streamsync run hello" without an internet connection and it seemed to work fine. Does anyone know if Streamsync requires an internet connection?

**Reply by HamsterSqueak** [reply]
My apologies. I did not see your question. I was using Visual Studio, not Visual Studio Code.
I will try docker to see if I can get that working. Thanks!
